src/provider_dota.ts

```diff
@@ -18,37 +18,45 @@ export default class DotaProvider extends Provider {
   }
 
   public async getNotifications(date?: Date, limit?: number): Promise<BotNotification[]> {
-    const pageDoc = await this.getPatchPage();
-    const patchList = await this.getPatchList(pageDoc);
-    let lastPatch = this.lastPatch;
-    const newPatches = [];
+    let notifications: BotNotification[] = [];
+    try {
+      const pageDoc = await this.getPatchPage();
+      const patchList = await this.getPatchList(pageDoc);
+      let lastPatch = this.lastPatch;
+      const newPatches = [];
 
-    // Discard the old patches
-    for (let i = 0; i < patchList.length && patchList[i] !== lastPatch; i++) {
-      newPatches.push(patchList[i]);
-    }
-
-    // Update the last patch version
-    if (newPatches.length > 0) {
-      lastPatch = newPatches[0];
-      this.setLastPatch(lastPatch);
-    }
+      // Discard the old patches
+      for (let i = 0; i < patchList.length && patchList[i] !== lastPatch; i++) {
+        newPatches.push(patchList[i]);
+      }
 
-    // Convert the patches to notifications
-    const notifications = newPatches.map((value) => {
-      return new BotNotification(
-        this.game,
-        'New gameplay update!',
-        new NotificationElement(`Gameplay patch ${value}`, `http://www.dota2.com/patches/${value}`),
-        `Gameplay patch ${value}`,
-        new Date(),
-        '',
-        '',
-        new NotificationElement('Dota 2'),
-      );
-    });
+      // Update the last patch version
+      if (newPatches.length > 0) {
+        lastPatch = newPatches[0];
+        this.setLastPatch(lastPatch);
+      }
 
-    return notifications;
+      // Convert the patches to notifications
+      notifications = newPatches.map((value) => {
+        return new BotNotification(
+          this.game,
+          'New gameplay update!',
+          new NotificationElement(
+            `Gameplay patch ${value}`,
+            `http://www.dota2.com/patches/${value}`,
+          ),
+          `Gameplay patch ${value}`,
+          new Date(),
+          '',
+          '',
+          new NotificationElement('Dota 2'),
+        );
+      });
+    } catch (error) {
+      botLogger.error('Dota updates page parsing failed');
```

Comment:
Shouldn't we log the error message here as well?

Reply:
This is also missing the logger label.

Reply:
fixed